Makefile

```diff
@@ -1,41 +1,96 @@
+# Makefile basic env setting
+.DEFAULT_GOAL := help
+
+
+# Makefile ARGS
 OR_EXEC ?= $(shell which openresty)
 LUA_JIT_DIR ?= $(shell ${OR_EXEC} -V 2>&1 | grep prefix | grep -Eo 'prefix=(.*)/nginx\s+--' | grep -Eo '/.*/')luajit
 LUAROCKS_VER ?= $(shell luarocks --version | grep -E -o  "luarocks [0-9]+.")
 LUA_PATH ?= ./lib/?.lua;./deps/lib/lua/5.1/?.lua;./deps/share/lua/5.1/?.lua;;
 LUA_CPATH ?= ./deps/lib/lua/5.1/?.so;;
 
 
-### help:         Show Makefile rules.
+# Makefile ENV
+ENV_OS_NAME ?= $(shell uname -s | tr '[:upper:]' '[:lower:]')
+ENV_RESTY   ?= LUA_PATH="$(LUA_PATH)" LUA_CPATH="$(LUA_CPATH)" resty
+
+# AWK patch for mawk
+ifneq ($(shell command -v gawk),)
+	ENV_HELP_AWK_RULE ?= '{ if(match($$0, /^\s*\#{3}\s*([^:]+)\s*:\s*(.*)$$/, res)){ printf("    make %-15s : %-10s\n", res[1], res[2]) } }'
+else
+	ENV_HELP_AWK_RULE := '{ if(match($$0, /^\#\#\#([^:]+):(.*)$$/)){ split($$0, res, ":"); gsub(/^\#\#\#[ ]*/, "", res[1]); _desc=$$0; gsub(/^\#\#\#([^:]+):[ \t]*/, "", _desc); printf("    make %-15s : %-10s\n", res[1], _desc) } }'
```

Comment:
If I am you, I will prompt that people should use gawk instead. mawk is faster. But it doesn't support UTF8 well and one day you will find the support of mawk should be dropped.

Reply:
Make sense, thanks for suggestion.

I add it for run test in container, I had change my mind. no necessary to increase maintenance costs